include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -2976,6 +2976,16 @@ struct OrtApi {
   */
   ORT_API2_STATUS(GetSparseTensorIndices, _In_ const OrtValue* ort_value, enum OrtSparseIndicesFormat indices_format, _Out_ size_t* num_indices, _Outptr_ const void** indices);
 
+  /** \brief Used for custom operators, gets the CUDA stream to use to launch the custom CUDA kernel     
+  *   \see ::OrtCustomOp
+  * \param[context] OrtKernelContext instance
+  * \param[out] Returns pointer to a CUDA stream that can be used to launch the custom CUDA kernel.
+  *             If retrieving the CUDA stream is not relevant (CUDA not enabled in the build, kernel partitioned to
+  *             some other EP), then a nullptr is returned as the output param.
+  *             Do not free or mutate the returned pointer as it refers to internal data owned by the underlying session.
```

Comment:
This looks dangerous.

Reply:
Not quite understanding how it jeopardizes the session object. AFAIK we can't return a const stream as it can't be used to launch a kernel. It is well documented in the API that the returned pointer shouldn't be used for anything other than launching a kernel. If the user still chooses to do something that isn't allowed and mis-uses the returned instance, it corrupts their session.

Reply:
There is very similar precedent here too - https://github.com/microsoft/onnxruntime/blob/1db21da5aeb93a4b08a92a40566a0c630dc2641b/include/onnxruntime/core/session/onnxruntime_c_api.h#L2973.